### PR TITLE
Update date picker templates in grid filtering

### PR DESCRIPTION
**Pull request by jackofdiamond5**
### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html [hunk 1]
```diff
@@ -183,29 +183,21 @@ <h6 class="igx-filter-empty__title">
...
-                                    [value]="
-                                        selectedColumn.formatter
-                                        ? (value | columnFormatter:selectedColumn.formatter)
-                                        : (value | date:selectedColumn.pipeArgs.format:undefined:grid.locale)
-                                    "
-                                    [readonly]="true"
-                                    [disabled]="!selectedColumn || !selectedCondition || (selectedColumn && selectedColumn.filters.condition(selectedCondition).isUnary)"/>
-                        </igx-input-group>
-                    </ng-template>
+                <igx-date-picker #datePicker *ngIf="selectedColumn && selectedColumn.dataType === 'date'"
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html:186]: The date picker looks different. It seems that the box input type is not applied. Also even when the date picker is disabled if you click it, it gets focused.
image

File: projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html [hunk 2]
```diff
@@ -183,29 +183,21 @@ <h6 class="igx-filter-empty__title">
...
+                    [displayDensity]="'compact'"
+                    [locale]="grid.locale"
+                    [outlet]="grid.outlet"
+                    [displayFormat]="selectedColumn.pipeArgs.format"
+                    [placeholder]="grid.resourceStrings.igx_grid_filter_row_date_placeholder"
+                    [disabled]="!selectedColumn || !selectedCondition
+                        || (selectedColumn && selectedColumn.filters.condition(selectedCondition).isUnary)">
+                    <!-- disable default icons -->
+                    <igx-picker-toggle></igx-picker-toggle>
+                    <igx-picker-clear></igx-picker-clear>
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html:200]: At first this looks like a strange way to disable the default icons, but on a second thought I guess it is ok.

File: projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html [hunk 3]
```diff
@@ -183,29 +183,21 @@ <h6 class="igx-filter-empty__title">
...
-                            <input #searchValueInput
-                                    igxInput
-                                    tabindex="0"
-                                    (keydown)="invokeClick($event)"
-                                    (click)="openDialog(dropDownTarget.element.nativeElement)"
-                                    [placeholder]="grid.resourceStrings.igx_grid_filter_row_date_placeholder"
-                                    autocomplete="off"
-                                    [value]="
-                                        selectedColumn.formatter
-                                        ? (value | columnFormatter:selectedColumn.formatter)
```
Comment (DiyanDimitrov) [inline projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html:202]: Do we support a `formatter` function for the value?

Comment (jackofdiamond5) [reply projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html:202]: Yes, we support the `formatter` function, it is now added to the templates. :)

**Review by DiyanDimitrov — APPROVED** [review]

**Review by DiyanDimitrov — APPROVED** [review]

File: projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html (hunk 2 above)
Comment (damyanpetev) [reply projects/igniteui-angular/src/lib/grids/filtering/advanced-filtering/advanced-filtering-dialog.component.html:203]: Admittedly not super intuitive, thus the comment. Still beats a separate template/prop to change those and another prop to hide each IMO :)

**Review by damyanpetev — APPROVED** [review]

**Pull request merged**